lucet-analyze/src/main.rs

```diff
@@ -459,57 +321,263 @@ fn print_summary(summary: ArtifactSummary) {
+    let mut imported_symbols = summary.imported_symbols.clone();
+    for import in module_data.import_functions() {
+        match module_data.function_info()[import.fn_idx.as_u32() as usize].name {
+            Some(name) => {
+                println!("  Internal name: {}", name);
+            }
+            None => {
+                println!("  No internal name");
+            }
+        }
+        println!("    Imported as: {}/{}", import.module, import.name);
+
+        // Remove from the imported_symbols list to not double-count imported functions
+        if let Some(idx) = imported_symbols.iter().position(|x| x == &import.name) {
+            imported_symbols.remove(idx);
+        }
+    }
+
+    if imported_symbols.len() > 0 {
+        println!("");
+        println!("  Other imported symbols (from ELF headers):");
+        for import in &imported_symbols {
+            println!("    {}", import);
+        }
+    }
+}
+
+fn print_summary(summary: ArtifactSummary) {
+    println!("Required Symbols:");
+    println!(
+        "  {:30}: {}",
+        "lucet_module_data",
+        exists_to_str(&summary.symbols.lucet_module_data)
+    );
+    println!(
+        "  {:30}: {}",
+        "lucet_module_data_len",
+        exists_to_str(&summary.symbols.lucet_module_data_len)
+    );
+    println!(
+        "  {:30}: {}",
+        "lucet_function_manifest",
+        exists_to_str(&summary.symbols.lucet_function_manifest)
+    );
+    println!(
+        "  {:30}: {}",
+        "lucet_function_manifest_len",
+        exists_to_str(&summary.symbols.lucet_function_manifest_len)
+    );
+    println!(
+        "  {:30}: {}",
+        "wasm_data_segments",
+        exists_to_str(&summary.symbols.wasm_data_segments)
+    );
+    println!(
+        "  {:30}: {}",
+        "wasm_data_segments_len",
+        exists_to_str(&summary.symbols.wasm_data_segments_len)
+    );
+
+    println!("\nModule data:");
+    match summary.load_module_data() {
+        Some(Ok(module_data)) => {
+            summarize_module_data(&summary, module_data);
+        }
+        Some(Err(e)) => {
+            println!("  ERROR: {}", e.to_string().red().bold());
+        }
+        None => {
+            println!("  MISSING SYMBOL:");
+            if summary.symbols.lucet_module_data.is_none() {
+                println!("  - {}", "lucet_module_data".red().bold());
+            }
+            if summary.symbols.lucet_module_data_len.is_none() {
+                println!("  - {}", "lucet_module_data_len".red().bold());
+            }
         }
-    } else {
-        println!("  {}", "MISSING!".red().bold());
     }
 
     println!("");
-    println!("Trap Manifest:");
-    if let Some(trap_manifest) = summary.trap_manifest {
-        for row in trap_manifest.records {
-            println!("  {:25} {} traps", row.func_name, row.trap_count);
-            println!("      {:?}", row.sites);
+    println!("Data Segments:");
+    if let Some(data_segments) = summary.data_segments {
```

Comment:
important question: are data segments still a thing? I didn't see where in `lucetc` data under these symbols would be emitted, and it _seems_ like this turned into sparse data. Is that a correct understanding? (cc @tyler ?)